client/option_advanced.go

```diff
@@ -206,6 +206,33 @@ func WithBoundHandler(h remote.BoundHandler) Option {
 	return Option{F: func(o *client.Options, di *utils.Slice) {
 		di.Push(fmt.Sprintf("AddBoundHandler(%T)", h))
 
-		o.RemoteOpt.AppendBoundHandler(h)
+		exist := false
+		switch h.(type) {
+		case remote.InboundHandler:
+			for _, inboundHandler := range o.RemoteOpt.Inbounds {
+				if inboundHandler == h.(remote.InboundHandler) {
+					exist = true
+					break
+				}
+			}
+		case remote.OutboundHandler:
+			for _, outboundHandler := range o.RemoteOpt.Outbounds {
+				if outboundHandler == h.(remote.OutboundHandler) {
+					exist = true
+					break
+				}
+			}
+		case remote.DuplexBoundHandler:
+			for _, duplexboundHandler := range o.RemoteOpt.Inbounds {
```

Comment:
In fact, DuplexBoundHandler is not necessary to be checked, because any DuplexBoundHandler also implements InboundHandler, you just need to check InboundHandler.